addons/mail/static/src/web/activity/activity.js

```diff
@@ -42,6 +43,9 @@ export class Activity extends Component {
             delay: computeDelay(this.props.data.date_deadline),
         });
         this.popover = usePopover(ActivityMarkAsDone, { position: "right" });
+        onChange(this.props.data, "date_deadline", () => {
+            this.state.delay = computeDelay(this.props.data.date_deadline);
```

Comment:
make `delay` a getter, call `render` in `updateDelayAtNight`, make sure its timeout is cleared when the component is unmounted/destroyed

Reply:
changes done!